### PR TITLE
🐛 source-postgres: fix jsonb[] handling bug

### LINKED ISSUE #29172: Postgres Source - Unable to handle jsonb[] (regression)
### Connector Name

destination-bigquery

### Connector Version

1.7.2

### What step the error happened?

During the sync

### Revelant information

Normalization: raw data (json)
Source: Postgres v3.0.2

In table A (at Postgres) we have a field `tags` (jsonb[]), so the data stored, looks like this: 
```
{"{\"key\": \"xxx_field\", \"value\": \"DEFAULT\", \"yyy_field\": true}","{\"key\": \"zzz_field\", \"value\": \"obfuscated\", \"rrr_field\": false}"}
``` 

Or like this:

```
{"{\"key\": \"xxx_field\", \"value\": \"DEFAULT\", \"yyy_field\": true}"}
```

The BigQuery Destination v2 connector, write the data in the `_airbyte_data ` like this: 

```
{
  "tags": "{\"{\\\"key\\\": \\\"origin\\\", \\\"value\\\": \\\"obfuscated\\\"}\"}"
}

```

In the final destination table, airbyte is unable to handle the `tags` field content, so it will be ignored with this message in the `_airbyte_meta `: 

{"errors":["Problem with `tags`"]}

**Expected bahaviour:**
Just set the field content in the final destination table.

### Relevant log output

_No response_

### Contribute

- [ ] Yes, I want to contribute

### LINKED ISSUE #29551: Source postgres: Cursor-based replication reads JSONB[] columns incorrectly
consider this setup:
```sql
create table jsonb_test (id serial primary key, q jsonb[]);
insert into jsonb_test (q) values (ARRAY['{"foo": "bar"}' :: jsonb]);
```

In CDC mode, source-postgres correctly emits a record with this data:
```json
{
  "q":["{\"foo\": \"bar\"}"]
}
```
Note that it's a JSON array of serialized JSON objects. (the serialization is up for debate as part of the JSONB project, and is not relevant to this issue).

However, in cursor-based mode, we read something like this:
```json
{
  "q": "{\"{\\\"foo\\\": \\\"bar\\\"}\"}"
}
```
Which looks like we took Postgres' string representation of an array (`{"{\"foo\":\"bar\"}"}`) and dumped it into a JSON string. This is wrong; we should parse out the array elements and build a proper json array.

This is a regression between v1.0.30 and v3.0.2; see https://github.com/airbytehq/airbyte/issues/29172#issuecomment-1683017904. I did a bisect and found that 1.0.42 is the last version where `jsonb[]` works correctly in non-cdc mode. (which is surprising, because 1.0.43's changelog notes don't mention anything about jsonb)

### PR DESCRIPTION
**Pull request by postamar**
This PR fixes https://github.com/airbytehq/airbyte/issues/29172 by adding a new subcase for unmatched `PostgresType`s in `copyToJsonField`. This new subcase distinguishes remaining array types from remaining non-array types. Added a unit test to cover.

Fixes https://github.com/airbytehq/airbyte/issues/29551.

### CODE REVIEW

**Comment by airbyte-oss-build-runner** [comment]
##  source-postgres test report (commit [`721bc36f71`](https://github.com/airbytehq/airbyte/pull/30534/commits/721bc36f71e176efde92417bde3dc57de0947909)) - ❌

⏲️  Total pipeline duration: 26mn57s 

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Build connector tar                                                    | ✅        |
| Build source-postgres docker image for platform linux/x86_64           | ✅        |
| Java Connector Unit Tests                                              | ✅        |
| Java Connector Integration Tests                                       | ✅        |
| Acceptance tests                                                       | ✅        |
| Validate airbyte-integrations/connectors/source-postgres/metadata.yaml | ✅        |
| Connector version semver check                                         | ✅        |
| Connector version increment check                                      | ❌        |
| QA checks                                                              | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/postamar_fix-29551/1695657977/721bc36f71e176efde92417bde3dc57de0947909/source-postgres/3.1.8/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:721bc36f71e176efde92417bde3dc57de0947909)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-postgres test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-postgres test report (commit [`851f593303`](https://github.com/airbytehq/airbyte/pull/30534/commits/851f5933038dbe90468b10602751192a448c7a66)) - ✅

⏲️  Total pipeline duration: 26mn38s 

| Step                                                                   | Result   |
|:-----------------------------------------------------------------------|:---------|
| Build connector tar                                                    | ✅        |
| Build source-postgres docker image for platform linux/x86_64           | ✅        |
| Java Connector Unit Tests                                              | ✅        |
| Java Connector Integration Tests                                       | ✅        |
| Acceptance tests                                                       | ✅        |
| Validate airbyte-integrations/connectors/source-postgres/metadata.yaml | ✅        |
| Connector version semver check                                         | ✅        |
| Connector version increment check                                      | ✅        |
| QA checks                                                              | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/postamar_fix-29551/1695674883/851f5933038dbe90468b10602751192a448c7a66/source-postgres/3.1.9/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:851f5933038dbe90468b10602751192a448c7a66)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-postgres test
```

**Review by nguyenaiden — COMMENTED** [review]
This looks good to me! Could you make sure to bump the version in `source-postgres-strict-encrypt` as well?

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java [hunk 1]
```diff
@@ -217,7 +217,13 @@ public void copyToJsonField(final ResultSet resultSet, final int colIndex, final
             case TIMESTAMP -> putTimestamp(json, columnName, resultSet, colIndex);
             case BLOB, BINARY, VARBINARY, LONGVARBINARY -> putBinary(json, columnName, resultSet, colIndex);
             case ARRAY -> putArray(json, columnName, resultSet, colIndex);
-            default -> json.put(columnName, value);
+            default -> {
+              if (columnInfo.columnType.isArrayType()) {
+                putArray(json, columnName, resultSet, colIndex);
```
Comment (rodireich) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java:222]: How come jsonb[] type not falling into `case ARRAY` ?

File: airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/AbstractPostgresSourceDatatypeTest.java [hunk 2]
```diff
@@ -847,6 +847,17 @@ private void addArraysTestData() {
...
 
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("jsonb_array")
+            .fullSourceDataType("JSONB[]")
+            .airbyteType(JsonSchemaType.builder(JsonSchemaPrimitive.ARRAY)
+                .withItems(JsonSchemaType.builder(JsonSchemaPrimitive.STRING).build())
+                .build())
+            .addInsertValues("ARRAY['{\"foo\":\"bar\"}'::JSONB, NULL]")
+            .addExpectedValues("[\"{\\\"foo\\\": \\\"bar\\\"}\",null]")
```
Comment (akashkulk) [inline airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/AbstractPostgresSourceDatatypeTest.java:858]: This still seem likes Postgres's string representation of the array. I think we want something along the lines of `["{\"foo\": \"bar\"}"]`

Comment (akashkulk) [reply airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/AbstractPostgresSourceDatatypeTest.java:858]: As discussed offline, disregard this. It's Java's representation of the array. We've tested this version manually and it fixes the regression 👍

**Review by akashkulk — APPROVED** [review]

**Pull request merged**